sklearn/decomposition/_nmf.py

```diff
@@ -1645,36 +1555,12 @@ def _check_params(self, X):
                 UserWarning,
             )
 
-        # alpha and regularization are deprecated in favor of alpha_W and alpha_H
-        # TODO clean up in 1.2
-        if self.alpha != "deprecated":
-            warnings.warn(
-                "`alpha` was deprecated in version 1.0 and will be removed "
-                "in 1.2. Use `alpha_W` and `alpha_H` instead",
-                FutureWarning,
-            )
-            alpha = self.alpha
-        else:
-            alpha = 0.0
-
-        if self.regularization != "deprecated":
-            warnings.warn(
-                "`regularization` was deprecated in version 1.0 and will be "
-                "removed in 1.2. Use `alpha_W` and `alpha_H` instead",
-                FutureWarning,
-            )
-            regularization = self.regularization
-        else:
-            regularization = "both"
-
         (
             self._l1_reg_W,
             self._l1_reg_H,
             self._l2_reg_W,
             self._l2_reg_H,
-        ) = _compute_regularization(
-            alpha, self.alpha_W, self.alpha_H, self.l1_ratio, regularization
-        )
+        ) = _compute_regularization(self.alpha_W, self.alpha_H, self.l1_ratio)
```

Comment:
It seems that this call to `_compute_regularization` can now be moved to `_scale_regularization` for simplicity, removing the need to `self._l1_reg_W` and co.

Actually, `_compute_regularization` and `_scale_regularization` would be better merged.

Reply:
I'm not sure we can merge both. At least we can't compute the scaled regularization only once in fit, because it would lead to a wrong scaling at transform. In MiniBatchNMF, the last batch may have a different size so we need to rescale.

I agree that it looks like there's room for simplification but I'm not able to see it. Maybe you have a clearer vision ?

Reply:
`_compute_regularization` only depends on fixed parameters, so storing the results does not seem necessary. I was thinking about removing `_compute_regularization`, and put its computation within `_scale_regularization`.

Reply:
Ah I see, I was think the other way around :)